src/FMBot.Bot/Commands/FriendsCommands.cs

```diff
@@ -173,65 +173,52 @@ public async Task AddFriends([Summary("Friend names")] params string[] friends)
             try
             {
                 var friendNotFoundList = new List<string>();
+                var duplicateFriendsList = new List<string>();
 
                 var addedFriendsCount = 0;
 
                 var existingFriends = await this._friendsService.GetFMFriendsAsync(this.Context.User);
 
                 if (existingFriends.Count + friends.Length > 10)
                 {
-                    await ReplyAsync("Sorry, but you can't have more then 10 friends.");
+                    await ReplyAsync("Sorry, but you can't have more than 10 friends.");
                     return;
                 }
 
                 foreach (var friend in friends)
                 {
-                    if (!existingFriends.Select(s => s.ToLower()).Contains(friend.ToLower()))
-                    {
-                        var friendUser = await this._guildService.FindUserFromGuildAsync(this.Context, friend);
+                    var friendGuildUser = await this._guildService.FindUserFromGuildAsync(this.Context, friend);
 
-                        if (friendUser != null)
-                        {
-                            var friendUserSettings = await this._userService.GetUserSettingsAsync(friendUser);
-
-                            if (friendUserSettings == null || friendUserSettings.UserNameLastFM == null)
-                            {
-                                if (await this._lastFmService.LastFMUserExistsAsync(friend))
-                                {
-                                    await this._friendsService.AddLastFMFriendAsync(this.Context.User.Id,
-                                        friend);
-                                    addedFriendsCount++;
-                                }
-                                else
-                                {
-                                    friendNotFoundList.Add(friend);
-                                }
-                            }
-                            else
-                            {
-                                await this._friendsService.AddDiscordFriendAsync(this.Context.User.Id, friendUser.Id);
-                                addedFriendsCount++;
-                            }
-                        }
-                        else
+                    Persistence.Domain.Models.User friendUserSettings;
+                    string username;
```

Comment:
`username` is a bit vague.  Maybe change it to `friendUsername` or something like that?